lib/wanda/results/execution_result.ex

```diff
@@ -5,15 +5,48 @@ defmodule Wanda.Results.ExecutionResult do
 
   use Ecto.Schema
 
+  import Ecto.Changeset
+
+  alias Wanda.Execution.Result
+
   @type t :: %__MODULE__{}
 
   @derive {Jason.Encoder, [except: [:__meta__]]}
   @primary_key false
   schema "execution_results" do
     field :execution_id, Ecto.UUID, primary_key: true
     field :group_id, Ecto.UUID
-    field :payload, :map
+    field :payload, :map, default: %{}
+
+    field :status, Ecto.Enum, values: [:running, :completed], default: :running
+
+    embeds_many :targets, Target do
+      field :agent_id, Ecto.UUID
+      field :checks, {:array, :string}
+    end
 
     timestamps(type: :utc_datetime_usec)
+
+    field :completed_at, :utc_datetime_usec
+  end
+
+  @spec changeset(t() | Ecto.Changeset.t(), map) :: Ecto.Changeset.t()
+  def changeset(execution, params) do
+    execution
+    |> cast(params, __MODULE__.__schema__(:fields) -- [:targets])
+    |> cast_embed(:targets, with: &target_changeset/2)
+  end
+
+  @spec complete(t(), Result.t()) :: Ecto.Changeset.t()
+  def complete(execution, %Result{} = result) do
```

Comment:
I am not sure about this. I have never found a schema that needs this kind of function since it already has the changeset. You can easily build a changeset in the context without having to rely to this one.

Reply:
agreed on renaming this to `complete_changeset`

Reply:
here https://github.com/trento-project/wanda/pull/60/commits/9ec12528f08b35acadd9b31c91f865877e111630